CompositionRoot/Factories/BFactory.swift

```diff
@@ -0,0 +1,16 @@
+protocol BFactoryProtocol {
+    func createB() -> BProtocol
+}
+
+class BFactory: BFactoryProtocol {
+    let applicationContext: ApplicationContext
+    
+    init(applicationContext: ApplicationContext) {
+        self.applicationContext = applicationContext
+    }
+    
+    func createB() -> BProtocol {
+        return B(cFactory: CFactory(applicationContext: applicationContext))
```

Comment:
We does not expect to reuse factory classes, so it can depend on any specific classes: `B` and `CFactory`.